app/uk/gov/hmrc/economiccrimelevyregistration/models/Registration.scala

```diff
@@ -40,6 +40,7 @@ final case class Registration(
   contactAddressIsUk: Option[Boolean],
   partnershipName: Option[String],
   privateBetaAccessCode: Option[String],
+  otherEntityJourneyData: OtherEntityJourneyData,
```

Comment:
I need to have a think about how we can release these without it being a breaking change for mid-flight journeys in production, as we have made this a 'mandatory' field in the model here, any in flight data will result in errors

we may have to do something in the backend whereby we populate this if it's not there, as a temporary 'migration' to prevent this being a potential breaking change

the risk is low as we only have a few users in live for private beta at the moment, but it's a risk none the less and would cause users to get stuck

Reply:
I think a solution might be the following:

- We make `otherEntityJourneyData` an `Option` on the `Registration` case class in the FE and BE.
- We add a helper method on the `Registration` case class that effectively 'removes' the optionality when trying to retrieve it, and throws an exception like throw new `llegalStateException("No other entity journey data in registration")`, we can use this when we are then getting/updating the data, as we should only be using this method in places where we expect the field to be present in the first place, and it'll be present after the initial question is answered.
- For the initial question about the other entity type, we can just check if it's there first and act accordingly

Let me know if you want to pair up on this / a call to explain

Reply:
Can't we make otherEntityJourneyData an option on the BE and leave it as is on the FE? We will obviously read/write it an an option on the FE when it goes to/from the BE.